users/clients/service.go

```diff
@@ -315,7 +315,7 @@ func (svc service) UpdateClientSecret(ctx context.Context, token, oldSecret, new
 		return mfclients.Client{}, err
 	}
 	if _, err := svc.IssueToken(ctx, dbClient.Credentials.Identity, oldSecret); err != nil {
-		return mfclients.Client{}, errors.ErrAuthentication
+		return mfclients.Client{}, errors.ErrSecretError
```

Comment:
This should return the `IssueToken` error, then we make IssueToken errors more sensible. 

Something like

```suggestion
		return mfclients.Client{}, err
```

of wrap the error

```suggestion
		return mfclients.Client{}, errors.Wrap(errors.ErrAuthentication, err)
```